### PR TITLE
feat: two dimensions phase correction filter

### PR DESCRIPTION
**Pull request by hamed-musallam**

### CODE REVIEW

File: src/component/reducer/actions/FiltersActions.ts [hunk 1]
```diff
@@ -76,7 +83,6 @@ type ZeroFillingFilterLiveAction = ActionType<
 type ManualPhaseCorrectionFilterAction = ActionType<
   | 'APPLY_MANUAL_PHASE_CORRECTION_FILTER'
   | 'CALCULATE_MANUAL_PHASE_CORRECTION_FILTER'
-  | 'APPLY_MANUAL_PHASE_CORRECTION_TOW_DIMENSION_FILTER'
   | 'CALCULATE_TOW_DIMENSIONS_MANUAL_PHASE_CORRECTION_FILTER',
```
[the remark this answers is not in the document]
Comment (hamed-musallam) [reply src/component/reducer/actions/FiltersActions.ts:86]: thanks Alejandro, i will remove it

[the remark this answers is not in the document]
Comment (hamed-musallam) [reply src/component/reducer/actions/FiltersActions.ts:86]: Oh !!!, i did the changes locally but i did not push the changes

**Comment by lpatiny** [comment]
I play with it seems to me quite intuitive. 2D phasing is never an easy task. I could be merge now and we can add issues after.

I would prefer a toggle button rather that a drop down menu to select Vertical / Horizontal

You could use 
image
 and 
image

**Comment by lpatiny** [comment]
There is an issue because we need to apply on the 2D the phase correction at least before switching between axes.

We could try to apply the 2D phase correction on mouse up (once we finished phasing PH0 and PH1). Same is true when we edit the value directly.

**Comment by lpatiny** [comment]
Seems weird PH0 and PH1 do not change when I switch the direction.



Also when we switch from axes it the phase applied on the 2D ?

**Comment by hamed-musallam** [comment]
@lpatiny 

When we change the pH values we apply them to the 2D. We debounce the applying on the 2D for 250 ms.

**Comment by lpatiny** [comment]
Debouncing is a good idea but if redrawing takes a long time it could still be problematic. We will see in the future if it is okay or if we should only recalculate when we release the pressed button on the button PH0 or PH1.

**Pull request merged**